src/herder/HerderPersistenceImpl.cpp

```diff
@@ -271,6 +271,60 @@ HerderPersistence::copySCPHistoryToStream(Database& db, soci::session& sess,
     return n;
 }
 
+optional<Hash>
+HerderPersistence::getNodeQuorumSet(Database& db, soci::session& sess,
+                                    NodeID const& nodeID)
+{
+    std::string nodeIDStrKey = KeyUtils::toStrKey(nodeID);
+    std::string qsethHex;
+
+    auto timer = db.getSelectTimer("quoruminfo");
+    soci::statement st = (sess.prepare << "SELECT qsethash FROM quoruminfo "
+                                          "WHERE nodeid = :id",
+                          soci::into(qsethHex), soci::use(nodeIDStrKey));
+
+    st.execute(true);
+
+    optional<Hash> res;
+    if (st.got_data())
+    {
+        auto h = hexToBin256(qsethHex);
+        res = make_optional<Hash>(std::move(h));
+    }
+    return res;
+}
+
+SCPQuorumSetPtr
+HerderPersistence::getQuorumSet(Database& db, soci::session& sess,
```

Comment:
From a separate-commits-compile-on-their-own perspective, I think you are calling this function in the commit before this one that defines it. I don't care _too_ much but if you want to clean up / rebase-edit, might want to fix that.

Reply:
fixed